tensorly/utils/jointdiag.py

```diff
@@ -0,0 +1,182 @@
+from itertools import combinations
+import tensorly as tl
+
+# Authors: Aaron Meyer <a@ameyer.me>
+#          Sebastian Seljak
+
+# License: BSD 3 clause
+
+
+def jointdiag(
```

Comment:
I know it's super verbose but I'd lean towards a clearer name, e.g. even `joint_matrix_diagonalization`.